src/main/java/core/basesyntax/operation/SupplyHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.transaction.FruitTransaction;
+
+public class SupplyHandler implements OperationHandler {
+    private ShopServiceImpl shopServiceImpl;

```

Comment:
Field `shopServiceImpl` is never initialized. You should initialize it through a constructor or a setter.